ext/storage_sources/gcp_store/test/test_gcp_connection.cpp

```diff
@@ -27,8 +27,261 @@
  */
 
 #define CATCH_CONFIG_MAIN
+#include "gcp_connection.h"
+
 #include <catch2/catch.hpp>
+#include <typeinfo>
 
-#include "gcp_connection.h"
+namespace gcs = google::cloud::storage;
+using namespace gcs;
+
+static std::string
+create_file(const std::string file_name, const std::string payload)
+{
+    std::ofstream file(file_name);
+    file << payload;
+    file.close();
+    return file_name;
+}
+
+static auto
+upload_file(gcs::Client client, std::string bucket_name, const std::string bucket_prefix,
+  const std::string file_name, const std::string object_name)
+{
+    auto metadata = client.UploadFile("./" + file_name, bucket_name, bucket_prefix + object_name);
+
+    return metadata;
+}
+
+static bool
+check_exist(gcs::Client client, std::string bucket_name, const std::string bucket_prefix,
+  const std::string object_name)
+{
+    auto metadata = client.GetObjectMetadata(bucket_name, bucket_prefix + object_name);
+
+    // Metadata ok implies that the file is present.
+    return metadata.ok();
+}
+
+static bool
+check_object_num(
```

Comment:
I'd recommend changing this function to something like `num_objects_in_bucket`, having it return the number, and then at the top level using `REQUIRE(num_objects_in_bucket() == 1)` for a bit more clarity.

Reply:
addressed here: [8039651](https://github.com/wiredtiger/wiredtiger/pull/8749/commits/8039651071462d3bd21d784fdb335f39875e742f)